relay/pkg/block/forwarder.go

```diff
@@ -56,20 +63,126 @@ func RunForwarder(
 		hostChain:    hostChain,
 		headersQueue: make(chan *btc.Header, headersQueueSize),
 		errChan:      make(chan error, 1),
+		quit:         make(chan bool, 1),
 	}
 
+	go forwarder.pullingLoop(ctx)
 	go forwarder.pushingLoop(ctx)
 
 	return forwarder
 }
 
+func (f *Forwarder) findBestBlock() (*btc.Header, error) {
+	currentBestDigest, err := f.hostChain.GetBestKnownDigest()
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Infof("best known digest returned from Ethereum network: %s",
+		hex.EncodeToString(currentBestDigest[:]),
+	)
+
+	bestHeader, err := f.btcChain.GetHeaderByDigest(currentBestDigest)
+	if err != nil {
+		return nil, err
+	}
+
+	betterOrSameHeader, err := f.btcChain.GetHeaderByHeight(bestHeader.Height)
+	if err != nil {
+		return nil, err
+	}
+
+	// see if there's a better block at that height
+	// if so, crawl backwards
+
+	// TODO: Is it ever possible that bestHeader and betterOrSameHeader are not
+	// equal?
```

Comment:
I think it may happen that:
1. We register new best header with digest X and height Y on the host chain
2. We crash soon due to an error
3. A reorg happens on BTC chain and height Y now corresponds to a digest Z 
4. We recover from the crash and initialize the pulling loop so we need to make sure what happened in the meantime on BTC chain

Reply:
That makes sense. Thanks.